mapper/tedge_mapper/src/sm_c8y_mapper/mapper.rs

```diff
@@ -27,7 +28,8 @@ impl TEdgeComponent for CumulocitySoftwareManagementMapper {
         let mqtt_config = mqtt_config(&tedge_config)?;
         let mqtt_client = Client::connect("SM-C8Y-Mapper", &mqtt_config).await?;
 
-        let sm_mapper = CumulocitySoftwareManagement::new(mqtt_client);
+        let mut sm_mapper = CumulocitySoftwareManagement::new(mqtt_client, tedge_config);
+        let () = sm_mapper.init().await?;
```

Comment:
why init is async here, can it run without init?

Reply:
This is a little tricky one, this `init` may fail and it seemed to be fair for it to be separated from the whole `run` method such that if you need to handle any error with it it would be easier, additionally it requires `&mut self` on the object to modify it (initialize `c8y_client_id` from external sources) and to not need to make entire `run` method `mut`.